py/server/deephaven/updateby_clause.py

```diff
@@ -0,0 +1,269 @@
+_JEmaControl = jpy.get_type("io.deephaven.api.updateby.EmaControl")
+_JMathContext = jpy.get_type("java.math.MathContext")
+
+
+class MathContext(Enum):
+    UNLIMITED = _JMathContext.UNLIMITED
+    """unlimited precision arithmetic"""
+
+    DECIMAL32 = _JMathContext.DECIMAL32
+    """a precision setting matching the IEEE 754R Decimal32 format, 7 digits"""
+
+    DECIMAL64 = _JMathContext.DECIMAL64
+    """a precision setting matching the IEEE 754R Decimal64 format, 16 digits"""
+
+    DECIMAL128 = _JMathContext.DECIMAL128
+    """a precision setting matching the IEEE 754R Decimal128 format, 34 digits"""
+
+
+class BadDataBehavior(Enum):
+    """An Enum defining ways to handle invalid data during EMA processing."""
+
+    RESET = _JBadDataBehavior.RESET
+    """Reset the state for the bucket to {@code null} when invalid data is encountered"""
+
+    SKIP = _JBadDataBehavior.SKIP
+    """Skip and do not process the invalid data without changing state"""
+
+    THROW = _JBadDataBehavior.THROW
+    """Throw an exception and abort processing when bad data is encountered"""
+
+    POISON = _JBadDataBehavior.POISON
+    """Allow the bad data to poison the result. This is only valid for use with NaN"""
+
+
+class EmaControl(JObjectWrapper):
+    """A EmaControl represents control parameters for performing EMAs (exponential moving average) with the table
+    update-by operation. """
+    j_object_type = _JEmaControl
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self.j_ema_control
+
+    def __init__(self, on_null: BadDataBehavior = BadDataBehavior.SKIP,
+                 on_nan: BadDataBehavior = BadDataBehavior.SKIP,
+                 on_null_time: BadDataBehavior = BadDataBehavior.SKIP,
+                 on_negative_deltatime: BadDataBehavior = BadDataBehavior.THROW,
+                 on_zero_deltatime: BadDataBehavior = BadDataBehavior.SKIP,
+                 big_value_context: MathContext = MathContext.DECIMAL128):
+        """Initializes a EmaControl for evaluating EMAs (exponential moving average) with table update-by operation.
+
+        Args:
+            on_null (BadDataBehavior): the behavior for when null values are encountered, default is SKIP
+            on_nan (BadDataBehavior): the behavior for when Nan values are encountered, default is SKIP
+            on_null_time (BadDataBehavior): the behavior for when null timestamps are encountered, default is SKIP
+            on_negative_deltatime (BadDataBehavior): the behavior for when negative sample-to-sample time differences are
+                encountered, default is THROW
+            on_zero_deltatime (BadDataBehavior): the behavior for when zero sample-to-sample-time differences are
+                encountered, default is SKIP
+            big_value_context (MathContext): the context to use when processing big numeric values (BigDecimal/BigInteger).
+                default is DECIMAL128.
```

Comment:
1) See note above about "context".
2) Is it best to call these "big numeric values" or "arbitrary precision numeric values".  I might lean to the latter, since that sounds like a type, while the current may be interpreted as the magnitude of a number (e.g. 1e6 is big).
3) Maybe a better name is `arb_precision_math`?